src/key.js

```diff
@@ -366,6 +366,27 @@ Key.prototype.getEncryptionKeyPacket = function() {
 };
 
 /**
+ * Encrypts all secret key and subkey packets
+ * @param  {String} passphrase
+ * @return {Boolean} true if all key and subkey packets encrypted successfully
+ */
+Key.prototype.encrypt = function(passphrase) {
+  if (this.isPrivate()) {
+    var keys = this.getAllKeyPackets();
+    for (var i = 0; i < keys.length; i++) {
+      try {
+        keys[i].encrypt(passphrase);
+      } catch (e) {
+        return false;
+      }
```

Comment:
you are right, i just followed what done in the decrypt function().

maybe it would be good to do the same there?